binhollc/SupernovaController
# 1223 | Update Dependencies in SupernovaController

**Resolves**

- https://focusuy.atlassian.net/browse/BMC2-1223

**How to test**

_Installation for end users_

- Create and activate Python environment with version 3.9.
- Suppress `GH_TOKEN` environment variable.
- Verify with `pip list` that none of `BinhoSupernova`, `hidapi`, `pyserial`, `supernovacontroller`, `transfer-controller` or `binhosimulators` packages are installed.
- Execute `pip install .`.
- Verify with `pip list` that all dependencies but `binhosimulators` have been installed.

_Installation for developers_

- Create and activate Python environment with version 3.9.
- Make sure that `GH_TOKEN` environment variable is set with the right credentials.
- Verify with `pip list` that none of `BinhoSupernova`, `hidapi`, `pyserial`, `supernovacontroller`, `transfer-controller` or `binhosimulators` packages are installed.
- Execute `pip install .`.
- Verify with `pip list` that all dependencies, including `binhosimulators` have been installed.

_Compatibility with Python 3.9 (test with simulated device)_

- Create and activate Python environment with version 3.9.
- Execute `pip install .`.
- Execute `python tests/test.py`.

Verify that all tests pass except for:

```
======================================================================
FAIL: test_spi_controller_transfer (__main__.TestSupernovaController)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gusa/Projects/Binho/SupernovaController/tests/test.py", line 517, in test_spi_controller_transfer
    self.assertTupleEqual((success, result), (True, [0x00, 0x04, 0x7F, 0x03, 0x02]))
AssertionError: Tuples differ: (True, b'\x9f\x00\x00\x00\x00') != (True, [0, 4, 127, 3, 2])

First differing element 1:
b'\x9f\x00\x00\x00\x00'
[0, 4, 127, 3, 2]

- (True, b'\x9f\x00\x00\x00\x00')
+ (True, [0, 4, 127, 3, 2])

----------------------------------------------------------------------
Ran 32 tests in 0.126s

FAILED (failures=1, skipped=1)
```

_Compatibility with Python 3.9 (test with simulated device)_

- Create and activate Python environment with version 3.9.
- Execute `pip install .`.
- Execute `USE_REAL_DEVICE=True python tests/test.py`.
- Verify that all tests pass.
